devito/types/basic.py

```diff
@@ -1019,29 +1038,33 @@ def _evaluate(self, **kwargs):
         This allow to evaluate off grid points as EvalDerivative that are better
         for the compiler.
         """
+        mapper = self._grid_map
+        subs = mapper.pop('subs', {})
         # Average values if at a location not on the Function's grid
-        if not self._grid_map:
+        if not mapper:
             return self
 
         io = self.interp_order
-        # Base function
         if self._avg_mode == 'harmonic':
-            retval = 1 / self.function
+            retval = 1 / self
         else:
-            retval = self.function
+            retval = self
 
         # Apply interpolation from inner most dim
-        for d, i in self._grid_map.items():
+        for d, i in mapper.items():
+            retval = retval._subs(i.subs(subs), self.indices_ref[d])
             retval = retval.diff(d, deriv_order=0, fd_order=io, x0={d: i})
 
         # Evaluate. Since we used `self.function` it will be on the grid when
         # evaluate is called again within FD
         retval = retval._evaluate(**kwargs)
+        if subs:
```

Comment:
I don't think this `if` is needed, `.subs` should take O(1) when `subs` is empty by returning immediately